lib/note-content-editor.tsx

```diff
@@ -205,9 +206,25 @@ class NoteContentEditor extends Component<Props> {
       } while ((node = node.next));
     };
     removeById(contextMenuLinks, removableIds);
+
+    const position = getNotePosition(noteId);
+
+    if (position) {
+      this.editor?.revealLineInCenter(position.line);
+      window.setTimeout(() => {
+        this.editor?.setScrollPosition({
```

Comment:
* I'm curious why we're calling both `revealLineInCenter` and `setScrollPosition`? `revealLineInCenter` [will scroll the note to the position requested](https://microsoft.github.io/monaco-editor/api/interfaces/monaco.editor.istandalonecodeeditor.html#reveallineincenter). Similarly, do we really need to save both the position and the `scrollTop` value?
* Did you test this without the timeout? It seems odd that we need it for `setScrollPosition` but not for `revealLineInCenter`.
* See also line 1106 where we use a promise that immediately resolves to update the cursor position; perhaps that would be faster than the 200ms timeout?

Reply:
You're right, there really isn't a need to keep both and call both. In this case I think keeping and setting the scroll position is the best way. The reason for this is the selected line may not match where you are scrolled to. Also if the line is long and wraps it will really only bring you to the top of that line and not where you were scrolled to.

The downside with scroll position and something I haven't accounted for yet is if the window size changes from the time when the position is saved and when you open the note again. There is a good chance it doesn't take you to the same spot in the note in this case.

Reply:
I've updated to use only scroll position now and clear the saved positions of things are resized as I think it is likely better to go to the top of a note rather than a random place if the size changes.